src/main/java/com/kodilla/ecommercee/controller/OrderController.java

```diff
@@ -1,35 +1,56 @@
 package com.kodilla.ecommercee.controller;
 
-import com.kodilla.ecommercee.domain.GenericEntity;
+import com.kodilla.ecommercee.domain.*;
+import com.kodilla.ecommercee.mapper.OrderMapper;
+import com.kodilla.ecommercee.service.OrderService;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
-
-import java.util.ArrayList;
 import java.util.List;
 
 @RestController
 @RequestMapping("/v1/order")
+@AllArgsConstructor
 public class OrderController {
+
+    @Autowired
+    private final OrderService orderService;
+    @Autowired
+    private final OrderMapper orderMapper;
+
     @GetMapping(value = "getOrders")
-    public List<GenericEntity> getOrders() {
-        return new ArrayList<>();
+    public List<OrderDto> getOrders() {
+        List<Order> orders = orderService.getAllOrders();
+        return orderMapper.mapToOrderDtoList(orders);
     }
 
     @PostMapping(value = "createOrder")
-    public void createOrder(GenericEntity genericEntity) {
+    public OrderDto createOrder(long cardId, long userId) throws CartNotFoundException, UserNotFoundException {
+        Cart cart = orderService.getCart(cardId).orElseThrow(CartNotFoundException::new);
+        User user = orderService.getUser(userId).orElseThrow(UserNotFoundException::new);
+        Order order = new Order(OrderStatus.UNPAID);
+        order.setCart(cart);
+        order.setUser(user);
+        Order savedOrder = orderService.saveOrder(order);
```

Comment:
dlaczego 29-34 nie robiłeś w serwisie?

Reply:
nie rozumiem pytania, w jakim sensie nie robiłem w serwisie?

Reply:
w sumie nie wiedziałem, że to ma być w serwisie. Ma to sens. Mogę przenieść, ale jesteś tego pewna, że tam to powinno być?

Reply:
Teraz tak patrzę, że zaoszczędze całe 3 linijki na tym 🙂

Reply:
ja jeszcze liczyłam wartośc zamówienia i ustawiałam to pole w Cart....

Reply:
tylko w kontrolerze też trzeba zrobić throws exception jak przeniesiesz do serwisu

Reply:
ok przeniosłem to częściowo